src/swift/Wrapper.swift

```diff
@@ -0,0 +1,283 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import CDispatch
+
+// This file contains declarations that are provided by the
+// importer via Dispatch.apinote when the platform has Objective-C support
+
+public class DispatchObject {
+	// TODO: add deinit method to invoke dispatch_release on wrapped()
+
+    internal func wrapped() -> dispatch_object_t {
+	    assert(false, "should be override in subclass")
+	}
+
+	public func setTarget(queue:DispatchQueue) {
+		dispatch_set_target_queue(wrapped(), queue.__wrapped)
+	}
+
+	public func activate() {
+		dispatch_activate(wrapped())
+	}
+
+	public func suspend() {
+		dispatch_suspend(wrapped())
+	}
+
+	public func resume() {
+		dispatch_resume(wrapped())
+	}
+}
+
+
+public class DispatchGroup : DispatchObject {
+	internal let __wrapped:dispatch_group_t;
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	public override init() {
+		__wrapped = dispatch_group_create()
+	}
+
+	public func enter() {
+		dispatch_group_enter(__wrapped)
+	}
+
+	public func leave() {
+		dispatch_group_enter(__wrapped)
+	}
+}
+
+public class DispatchSemaphore : DispatchObject {
+	internal let __wrapped: dispatch_semaphore_t;
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	public init(value: Int) {
+		__wrapped = dispatch_semaphore_create(value)
+	}
+}
+
+public class DispatchIO : DispatchObject {
+	internal let __wrapped:dispatch_io_t
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	internal init(__type: UInt, fd: Int32, queue: DispatchQueue,
+				  handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create(__type, fd, queue.__wrapped, handler)
+	}
+
+	internal init(__type: UInt, path: UnsafePointer<Int8>, oflag: Int32,
+				  mode: mode_t, queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_path(__type, path, oflag, mode, queue.__wrapped, handler)
+	}
+
+	internal init(__type: UInt, io: DispatchIO,
+				  queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_io(__type, io.__wrapped, queue.__wrapped, handler)
+	}
+
+	internal init(queue:dispatch_queue_t) {
+		__wrapped = queue
+	}
+
+	public func barrier(execute: () -> ()) {
+		dispatch_io_barrier(self.__wrapped, execute)
+	}
+
+	// FIXME: getter:DispatchIO.fileDescriptor(self:) ==> dispatch_io_get_descriptor
```

Comment:
fixed.  nothing deep; I'm still learning Swift and my first attempt botched the syntax for a computed property.